presto-main/src/main/java/com/facebook/presto/execution/StageExecutionStateMachine.java

```diff
@@ -369,16 +372,16 @@ public void recordGetSplitTime(long startNanos)
         runtimeStats.addMetricValue(GET_SPLITS_TIME_NANOS, NANO, elapsedNanos);
     }
 
-    public void recordSchedulerRunningTime(long cpuTimeNanos, long wallTimeNanos)
+    public void recordSchedulerRunningTime(long cpuTimeNanos, long wallTimeNanos, boolean isScanStage)
     {
-        runtimeStats.addMetricValue(SCHEDULER_CPU_TIME_NANOS, NANO, max(cpuTimeNanos, 0));
-        runtimeStats.addMetricValue(SCHEDULER_WALL_TIME_NANOS, NANO, max(wallTimeNanos, 0));
+        runtimeStats.addMetricValue(isScanStage ? SCAN_STAGE_SCHEDULER_CPU_TIME_NANOS : SCHEDULER_CPU_TIME_NANOS, NANO, max(cpuTimeNanos, 0));
```

Comment:
SCHEDULER_CPU_TIME_NANOS name is confusing now for what it represents. I would expect it to include the scheduling stage time as well.  If anyone is relying on this value they also may not realize that it now doesn't include the full scan stage. should we change the name or should we include the scan stage time here as well?

Reply:
This metric is introduced recently. So i dont think anyone is using,relying on it yet. 

So this schedulerCpuTime is a metric for every stage. So i want just distinguish the metric for scan stages, so that its easy to track. I did not completely understand your question.

Reply:
previously scheduler_cpu_time_nanos was the scheduler cpu time for all stages.  With this change, it is the scheduler cpu time for all stages except the scan stage. I think it would be better to either give the metric a name that makes it clear that it's the scheduler time for non-scan stages only, or have it also include the time for scan stages.  I don't have a preference about which way to go here.

Reply:
I see your point. I made changes so that for leaf scan stage we have both  `schedulerCpuTimeNanos`, `scanStageSchedulerCpuTimeNanos`